File: 151003/Shafarenko/lab5/redis/Discussion/PostEntity/Post.cs
using Discussion.Common;

namespace Discussion.PostEntity
{
    public class Post(string content, string country) : AbstractEntity
    {
        public string Country { get; set; } = country;
        public int NewsId { get; set; }
        public string Content { get; set; } = content;

        public Post() : this(string.Empty, string.Empty) { }

        public Post(int id, int newsId, string content) : this(content, string.Empty)
        {
            Id = id;
            NewsId = newsId;
        }

        public Post(int id, int newsId, string content, string country) : this(content, country)
        {
            Id = id;
            NewsId = newsId;
        }
    }
}

File: 151002/Luschickij/fullProject/DC_lab/discussion/src/main/java/com/luschickij/discussion/model/Post.java
package com.luschickij.discussion.model;


import edu.umd.cs.findbugs.annotations.DefaultAnnotation;
import jakarta.persistence.Entity;
import lombok.*;
import org.springframework.data.cassandra.core.cql.Ordering;
import org.springframework.data.cassandra.core.cql.PrimaryKeyType;
import org.springframework.data.cassandra.core.mapping.PrimaryKeyColumn;
import org.springframework.data.cassandra.core.mapping.Table;


@Table(value = "tbl_post")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Builder
public class Post extends IdentifiedEntity {


    @PrimaryKeyColumn(name = "country",
            type = PrimaryKeyType.PARTITIONED)
    private String country;

    @PrimaryKeyColumn(name = "newsid",
            ordinal = 0,
            type = PrimaryKeyType.CLUSTERED)
    private Long newsId;

    @PrimaryKeyColumn(
            name = "id",
            ordinal = 1,
            type = PrimaryKeyType.CLUSTERED)
    private Long id;

    private String content;
}
